google/cloud/internal/opentelemetry.h

```diff
@@ -94,6 +95,19 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
     opentelemetry::trace::StartSpanOptions const& options =
         DefaultStartSpanOptions());
 
+/**
+ * Start a span with a @p name and @p attributes using templates.
+ */
+template <class T>
```

Comment:
We would typically say `typename T`.

Reply:
done, and done for other places in the file